src/main/java/com/modsen/software/ratings/kafka/util/RideDeserializer.java

```diff
@@ -0,0 +1,33 @@
+package com.modsen.software.ratings.kafka.util;
+
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.modsen.software.ratings.kafka.event.BaseRideEvent;
+import com.modsen.software.ratings.kafka.event.SelectionDriverEvent;
+import com.modsen.software.ratings.kafka.event.StatusChangedEvent;
+import org.apache.kafka.common.serialization.Deserializer;
+
+import java.util.Map;
+
+public class RideDeserializer implements Deserializer<BaseRideEvent> {
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
+    @Override
+    public BaseRideEvent deserialize(String topic, byte[] data) {
```

Comment:
As far as I remember, there is mechanism to serialize and deserialize objects without such approach.
Just for knowledge, check it please